### PR TITLE
Fix rust warnings

### PR DESCRIPTION
**Pull request by aknuds1**
I got a lot of warnings after upgrading the Rust toolchain to 1.37, this PR fixes them via `cargo fix`. Also removed a couple of unused functions by hand, since `cargo fix` didn't take care of them.

### CODE REVIEW

File: server/prisma-rs/migration-engine/connectors/sql-migration-connector/src/database_inspector/mod.rs [hunk 1]
```diff
@@ -26,7 +26,7 @@ pub trait DatabaseInspector: Send + Sync + 'static {
     fn introspect(&self, schema: &String) -> DatabaseSchema;
 }
 
-impl DatabaseInspector {
```
Comment (pimeys) [inline server/prisma-rs/migration-engine/connectors/sql-migration-connector/src/database_inspector/mod.rs]: You could maybe move all of these methods directly to the trait.

Comment (aknuds1) [reply server/prisma-rs/migration-engine/connectors/sql-migration-connector/src/database_inspector/mod.rs]: @pimeys Could we do that in another PR? I tried, but it leads to non-trivial build failures.

Comment (pimeys) [reply server/prisma-rs/migration-engine/connectors/sql-migration-connector/src/database_inspector/mod.rs]: Can you show me what kind of failures? You're doing God's work here btw :)

Comment (aknuds1) [reply server/prisma-rs/migration-engine/connectors/sql-migration-connector/src/database_inspector/mod.rs]: @pimeys OK, I'll try to reproduce

Comment (aknuds1) [reply server/prisma-rs/migration-engine/connectors/sql-migration-connector/src/database_inspector/mod.rs]: @pimeys After discussing with Marcus, we'll merge this now and handle the trait in a follow-up PR. I'll get back to you later on this, OK? Got to handle another issue now.

**Comment by pimeys** [comment]
Thank you :)

File: server/prisma-rs/migration-engine/connectors/sql-migration-connector/src/database_inspector/mod.rs (hunk 1 above)
Comment (aknuds1) [reply server/prisma-rs/migration-engine/connectors/sql-migration-connector/src/database_inspector/mod.rs]: See https://github.com/prisma/prisma/pull/4836.

**Pull request merged**